src/magnify/registry.py

```diff
@@ -29,11 +32,254 @@ def component_factory(*args, **kwargs):
+        - A sequence (list or tuple) of file paths, `xarray.DataArray`, or `xarray.Dataset`.
+        The function can handle multiple formats and will standardize them into an `xarray.Dataset` for processing.
+    times : 
+        A list or sequence of time points to extract from the dataset. If specified, only the data
+        corresponding to these time points will be processed. If `None`, all available time points
+        in the data will be used.
+    channels :
+        A list or sequence of channel names to extract from the dataset. If specified, only the data
+        from the specified channels will be processed. If `None`, all available channels will be used.
+    debug :
+        If True, set the logger to debug mode for error logging.
+    shape : tuple[int, int], optional
+        The shape of the button array, specifying the number of rows and columns in the image grid.
+        If provided, it will be used to assign default button tags. The default is `(8, 8)` for mini chips.
+    pinlist :
+        A file path to a CSV file that describes the pin layout on the chip. The CSV file must include
+        a column called `Indices` that contains row and column pairs in the format `(row, col)`, and a
+        `MutantID` column that contains the names of the buttons. This is used to map the buttons to physical
+        locations on the chip. Either `pinlist` or `shape` must be provided.
+    blank : 
+        Values representing "blank" or non-expressed buttons in the pinlist, which will be replaced
+        with empty strings in the dataset. Defaults to `["", "blank", "BLANK"]`.
+    overlap :
+        The number of pixels to exclude from the edges of adjacent tiles during the stitching process.
+        This overlap value is subtracted from both the vertical (y) and horizontal (x) dimensions
+        of the tiles to remove redundant or overlapping areas between adjacent tiles.
+    rotation : 
+        The degree of rotation to apply to the stitched image.
+    row_dist / col_dist: 
+        The distance between rows/columns of buttons in pixels. This is converted to pixels based on the pixel-to-micron conversion rate, assuming 1.61 pixels.
+    min_button_radius / max_button_radius : 
+        The minimum/maximum radius (in pixels) for detecting buttons in the image.
+    low_edge_quantile / high_edge_quantile : 
+        The lower/upper quantile for edge detection, used to identify the dimmest edges when detecting buttons.
+    num_iter : 
+        The maximum number of iterations to perform during the button detection process, controlling the depth of search.
+    min_roundness : 
+        The minimum roundness value for detected buttons. Buttons that do not meet this roundness threshold are excluded.
+    cluster_penalty : 
+        The penalty factor for clustering during button detection. This parameter helps control how strongly nearby buttons are grouped together.
```

Comment:
Not 100% accurate the cluster penalty ends up in this line of code:
https://github.com/FordyceLab/magnify/blob/main/src/magnify/find.py#L628

Where I'm trying to find rows/columns of buttons and I'm saying that each cluster gets penalized for two things:
1) having a high inter-cluster variance (so the points are far away from the mean value)
2) not having the expected number of values inside the cluster, e.g. I expect to find 8 buttons per row but my cluster has 16 or 0 which would be bad.

A higher value for cluster_penalty increases how much I prioritize 2 vs 1, where a high value of cluster_penalty will mean that 2 is much more important. In practice I haven't found the need to really tune this value though and it's tough to explain in a docstring so we can be a bit handwavey with our explanation. :)

Reply:
@Karl-Krauth I have written cluster_penalty as follows. Please feel free to let me know if you have any suggestions:
```
    cluster_penalty : 
        A penalty number that balances two factors when identifying clusters: penalizing high inter-cluster variance and 
        penalizing deviations from the expected number of items in a cluster. A higher value places more emphasis on the second factor
```